pkg/openstack/memcached.go

```diff
@@ -199,6 +201,44 @@ func reconcileMemcached(
 		}
 
 		tlsCert = certSecret.Name
+
+		// mTLS cert
+		if spec.TLS.MTLS.SslVerifyMode == "Request" || spec.TLS.MTLS.SslVerifyMode == "Require" {
+			Log.Info("Reconciling Memcached mTLS", "Memcached.Namespace", instance.Namespace, "Memcached.Name", name)
+			//clusterDomain = clusterdns.GetDNSClusterDomain()
+			certRequest = certmanager.CertificateRequest{
+				IssuerName: instance.GetInternalIssuer(),
+				CertName:   fmt.Sprintf("%s-mtls", memcached.Name),
+				Hostnames: []string{
+					fmt.Sprintf("*.openstack.svc"),
+					fmt.Sprintf("*.openstack.svc.cluster.local"),
```

Comment:
lets use `clusterdns.GetDNSClusterDomain()` instead of hard coding `cluster.local` as in https://github.com/openstack-k8s-operators/openstack-operator/blob/main/pkg/openstack/common.go#L210